### PR TITLE
upgrade to rails 6.1

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made?

keep things well maintained

## How was this change tested?

unit tests (will also keep an eye on stage and prod after deployment)

## Which documentation and/or configurations were updated?

lots of rails configs

### CODE REVIEW

File: config/initializers/filter_parameter_logging.rb [hunk 1]
```diff
@@ -3,4 +3,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += %i[
+  password passw secret token _key crypt salt certificate otp ssn
```
Comment (jmartin-sul) [inline config/initializers/filter_parameter_logging.rb:7]: the new default file from rails omitted `password`.  i wasn't sure if that was already a default in `filter_parameters`, so i added it here to be safe, but happy to remove if it's redundant.

File: config.ru [hunk 2]
```diff
@@ -5,3 +5,4 @@
 require_relative 'config/environment'
 
 run Rails.application
```
Comment (jcoyne) [inline config.ru:7]: I don't think we need both of these lines, do we?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: config.ru (hunk 2 above)
Comment (jcoyne) [reply config.ru:7]: Looks like we do in fact need both.

File: config/application.rb [hunk 3]
```diff
@@ -1,7 +1,5 @@
 # frozen_string_literal: true
 
-require_relative 'boot'
```
Comment (jcoyne) [inline config/application.rb:3]: I'm pretty sure we need this line: https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/application.rb.tt#L1

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: config.ru (hunk 2 above)
Comment (jmartin-sul) [reply config.ru:7]: ah yeah, wasn't sure about that.  preference for which to keep?

File: config/application.rb (hunk 3 above)
Comment (jmartin-sul) [reply config/application.rb:3]: i was also surprised by the disappearance of that line, i'm pretty sure this came from the `rails app:update` invocation, but also could've been bad merging on my part.  will add back...

Comment (jcoyne) [reply config/application.rb]: It is there in a newly generated rails 6.1, so presumably we need it.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**